### PR TITLE
Implement official server map indexes.

### LINKED ISSUE #584: Correlate Map Indexes with Original Servers
## Summary
There are couple of hints in the client code that indicate that original map numbers may not line up with what would be expected based upon level requirements or even map name. An example of this is the internal number of 29 for Galaxy City, which has a map name of City_01_03 and is one of the starting zones.

### Proposal
The community has data-mined nearly all the original map numbers while the game was still live. I propose we utilize those numbers in SEGS. The benefits of this may not be readily apparent, but it's early enough in our development that the change wont have any real impact.

The easiest place to find the original zone indexes is in the zone.cfg and travel.cfg files used with Paragon Chat. I've attached them here.

P.S. in the future we can create scripts to be read from, similar to these cfgs here.
[zone.cfg.txt](https://github.com/Segs/Segs/files/2402836/zone.cfg.txt)
[travel.cfg.txt](https://github.com/Segs/Segs/files/2402837/travel.cfg.txt)

### Tasks to Complete This Issue
- [ ] Re-index the maps in `map_definitions.cpp` to the official numbers pulled from zone.cfg (linked above) 
- [ ] Exclude, or comment out, maps that our client version doesn't have. The existing list in `map_definitions.cpp` should be definitive (missions maps aside)
- [ ] Test to verify we didn't bork something

**Example: **
```cpp
{3, "City_01_03", "maps/City_Zones/City_01_03/City_01_03.txt", "Galaxy City"},
// would become:
{29, "City_01_03", "maps/City_Zones/City_01_03/City_01_03.txt", "Galaxy City"},
```

### Bonus Tasks that would make you the most awesome-est person evah!
- Add spawn indicator name to the vector in `map_definitions.cpp` and leverage it for `/stuck` or zoning
- Move `g_defined_map_deltas` vector to some sort of flat file, either json, lua, (yaml?) or cfg, that we can read from on loading the server. This will make server-customization and (future) custom maps possible
- Utilize the attached `travel.cfg` to generate a list of zones for `/mapmenu`

### PR DESCRIPTION
**Pull request by Cynicide**
Update g_defined_map_datas with original server indexes
Update methods in map_definitions to use new indexes
Update starting zone id in EntityData
Update map id types to uint32_t as it was chopping large map id's

## Summary
Updated `g_defined_map_datas` to correlate map indexes with indexes used on the original servers and modified relevant functions to use. 

Tested build locally by creating new characters, using /mapmenu to change maps, using /mz to change maps, providing /mz with incorrect data to test failures. 

Was unable to complete any bonus tasks.

### Issues closed
Closes #584

### Additions/modifications proposed in this pull request:
- Updated `g_defined_map_datas` to correlate map indexes with indexes used on the original servers.
- Updated `getMapIndex()`, `getDisplayMapName()`, `getMapName()` and `getMapPath()` to search for or use new indexes and return the Outbreak data if it is not found. 
- Updated `EntityData` with the new Outbreak index as the old index was hard coded
- Updated map index type in multiple locations from uint8_t to uint32_t to match type in `g_defined_map_datas` as it was chopping large map id's (eg. Elysium ID: 9928). **Please note that this change was in an area that may have been beyond my ability and should be checked.**

### CODE REVIEW

**Comment by Magentah** [comment]
Looks fine to me. I'd suggest making the default map index (24) a constant so that's it's clear - something like `const uint32_t DEFAULT_MAP_IDX = 24; // Outbreak` maybe.

File: Projects/CoX/Common/GameData/map_definitions.cpp [hunk 1]
```diff
@@ -6,34 +6,35 @@
...
-    {13, "Hazard_03_01", "maps/City_Zones/Hazard_03_01/Hazard_03_01.txt", "Dark Astoria"},
-    {14, "Hazard_04_01", "maps/City_Zones/Hazard_04_01/Hazard_04_01.txt", "Crey's Folly"},
-    {15, "Hazard_04_02", "maps/City_Zones/Hazard_04_02/Hazard_04_02.txt", "Enviro Nightmare"},
-    {16, "Hazard_05_01", "maps/City_Zones/Hazard_05_01/Hazard_05_01.txt", "Elysium"},
+    {12, "Hazard_01_01", "maps/City_Zones/Hazard_01_01/Hazard_01_01.txt", "Perez Park"},
+    {13, "Hazard_02_01", "maps/City_Zones/Hazard_02_01/Hazard_02_01.txt", "Boomtown"},
+    {43, "Hazard_03_01", "maps/City_Zones/Hazard_03_01/Hazard_03_01.txt", "Dark Astoria"},
+    {15, "Hazard_04_01", "maps/City_Zones/Hazard_04_01/Hazard_04_01.txt", "Crey's Folly"},
+    // Commented out as a dev zone with no objects.
+    //{"Unknown", "Hazard_04_02", "maps/City_Zones/Hazard_04_02/Hazard_04_02.txt", , "Enviro Nightmare"},
```
[the remark this answers is not in the document]
Comment (Magentah) [reply Projects/CoX/Common/GameData/map_definitions.cpp]: The text files in the issue don't include Enviro already, so we'd probably need to give it our own number. There does to be some sort of system to the indexes though (mission maps all seem to be together, unique maps all seem to be together, etc)

[the remark this answers is not in the document]
Comment (broxen) [reply Projects/CoX/Common/GameData/map_definitions.cpp]: Yeah I noticed one of the maps starts `99` and my guess is that those are maps PChat has created IDXs for.

[the remark this answers is not in the document]
Comment (Cynicide) [reply Projects/CoX/Common/GameData/map_definitions.cpp]: Could we claim 50000-59999? the indexes at the moment go up to 42000.

[the remark this answers is not in the document]
Comment (broxen) [reply Projects/CoX/Common/GameData/map_definitions.cpp]: I don't see any reason why not

[the remark this answers is not in the document]
Comment (Cynicide) [reply Projects/CoX/Common/GameData/map_definitions.cpp]: OK, done. Tested in /mz and /mapmenu.

**Pull request merged**